packages/compartment-mapper/src/capture-lite.js

```diff
@@ -79,12 +85,158 @@ const captureCompartmentMap = (compartmentMap, sources) => {
+      }
+
+      const compartment = compartments[compartmentName];
+      if (!compartment) {
+        throw new ReferenceError(
+          `No compartment found for ${q(compartmentName)}`,
+        );
+      }
+      const compartmentOwnModuleDescriptor =
+        compartmentDescriptor.modules[compartmentDescriptor.name];
+
+      if (!compartmentOwnModuleDescriptor?.module) {
+        throw new Error(
+          `Cannot determine entry point of ${q(compartmentName)}`,
+        );
+      }
+      acc.push([
+        compartmentName,
+        compartment,
+        compartmentOwnModuleDescriptor.module,
+      ]);
+
+      return acc;
+    }, /** @type {[compartmentName: string, compartment: Compartment, moduleSpecifier: string][]} */ ([]));
+
+    const { length: compartmentsToLoadCount } = compartmentsToLoad;
+    /**
+     * This index increments in the order in which compartments finish
+     * loading—_not_ the order in which they began loading.
+     */
+    let loadedCompartmentIndex = 0;
+    await Promise.all(
+      compartmentsToLoad.map(
+        async ([compartmentName, compartment, moduleSpecifier]) => {
+          await compartment.load(moduleSpecifier);
+          log(
+            `Force-loaded Compartment: ${q(compartmentName)} (${(loadedCompartmentIndex += 1)}/${compartmentsToLoadCount})`,
+          );
+        },
+      ),
+    );
+  };
+
+  /**
+   * Loads, in order:
+   *
+   * 1. The entry compartment
+   * 2. The attenuators compartment (_if and only if_ `policy` was provided)
+   * 3. All compartments in the `compartmentMap` that have the `load` bit set.
+   *
+   * @param {Record<string, Compartment>} linkedCompartments
+   * @param {Compartment} entryCompartment
+   * @param {Compartment} attenuatorsCompartment
+   * @returns {Promise<void>} Resolves when all compartments are loaded.
+   */
+  const loadCompartments = async (
+    linkedCompartments,
+    entryCompartment,
+    attenuatorsCompartment,
+  ) => {
+    await entryCompartment.load(entryModuleSpecifier);
+
+    if (policy) {
+      // retain all attenuators.
+      await Promise.all(
+        detectAttenuators(policy).map(attenuatorSpecifier =>
+          attenuatorsCompartment.load(attenuatorSpecifier),
+        ),
+      );
+    }
+
+    await forceLoadCompartments(linkedCompartments);
+  };
+
+  return loadCompartments;
+};
+
+/**
+ * "Captures" the compartment map descriptors and sources from a partially
+ * completed compartment map—_without_ creating an archive.
+ *
+ * The resulting compartment map represents a well-formed dependency graph,
+ * laden with useful metadata. This, for example, could be used for automatic
+ * policy generation.
+ *
+ * @param {ReadFn | ReadPowers} readPowers Powers
  * @param {CompartmentMapDescriptor} compartmentMap
  * @param {CaptureLiteOptions} [options]
  * @returns {Promise<CaptureResult>}
  */
-export const captureFromMap = async (powers, compartmentMap, options = {}) => {
+export const captureFromMap = async (
```

Comment:
besides the refactors, I renamed:

- parameter `powers` ➡️ `readPowers`
- internal name of option `Compartment` ➡️ `CompartmentOption`
- module-scoped `defaultCompartment` ➡️ `DefaultCompartment`